frontend/src/components/TradeBox/EncryptedChat/EncryptedTrutleChat/index.tsx

```diff
@@ -0,0 +1,303 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, Tooltip, TextField, Grid, Container, Paper, Typography } from '@mui/material';
+import { encryptMessage, decryptMessage } from '../../../../pgp';
+import { saveAsJson } from '../../../../utils';
+import { AuditPGPDialog } from '../../../Dialogs';
+import { systemClient } from '../../../../services/System';
+import { websocketClient, WebsocketConnection } from '../../../../services/Websocket';
+
+// Icons
+import CircularProgress from '@mui/material/CircularProgress';
+import KeyIcon from '@mui/icons-material/Key';
+import { ExportIcon } from '../../../Icons';
+import { useTheme } from '@mui/system';
+import MessageCard from '../MessageCard';
+import ChatHeader from '../ChatHeader';
+import { EncryptedChatMessage, ServerMessage } from '..';
+import { apiClient } from '../../../../services/api';
+import ChatBottom from '../ChatBottom';
+
+interface Props {
+  orderId: number;
+  userNick: string;
+  takerNick: string;
+  chatOffset: number;
+  messages: EncryptedChatMessage[];
+  setMessages: (messages: EncryptedChatMessage[]) => void;
+}
+
+const EncryptedTurtleChat: React.FC<Props> = ({
```

Comment:
Very clean solution for the 2 modes and the re-utilization of header and bottom buttons!